packages/docusaurus/src/client/routeAnnouncer.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React from 'react';
+import {useLocation} from 'react-router-dom';
+
+export function RouteAnnouncer() {
+  const {pathname} = useLocation();
+  const [routeAnnouncement, setRouteAnnouncement] = React.useState('');
+
+  // Only announce the path change, but not for the first load because screen
+  // reader will do that automatically.
+  const previouslyLoadedPath = React.useRef(pathname);
+
+  // Every time the path changes, announce the new page’s title following this
+  // priority: first the document title (from head), otherwise the first h1, or
+  // if none of these exist, then the pathname from the URL. This methodology is
+  // inspired by Marcy Sutton’s accessible client routing user testing. More
+  // information can be found here:
+  // https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing/
+  React.useEffect(
+    () => {
+      // If the path hasn't change, we do nothing.
+      if (previouslyLoadedPath.current === pathname) {
+        return;
+      }
+      previouslyLoadedPath.current = pathname;
+
+      if (document.title) {
+        setTimeout(() => {
+          setRouteAnnouncement(document.title);
+        }, 50);
+      } else {
+        const pageHeader = document.querySelector('h1');
+        const content = pageHeader?.innerText ?? pageHeader?.textContent;
+        setTimeout(() => {
+          setRouteAnnouncement(content || pathname);
+        }, 50);
```

Comment:
This timeout needs to be cleaned up. Also, we can simply do `document.title ?? content ?? pathname` I believe?

Reply:
I did a commit for `document.title ?? content ?? pathname`. But, unfortunately when I don't include the timeout the announcer will keep the previous content (in my testing).

Reply:
That's expected, because our route components are lazy-loaded with react-loadable, so when the router calls `ComponentCreator`, the new component hasn't mounted yet. We may have to do some refactors to that as I'm not really sure this timeout will work in all cases (e.g. if rendering is blocked)